src/Cli/dotnet/Commands/Build/BuildCommand.cs

```diff
@@ -35,6 +34,8 @@ public static CommandBase FromParseResult(ParseResult parseResult, string msbuil
 
         string[] forwardedOptions = parseResult.OptionValuesToBeForwarded(BuildCommandParser.GetCommand()).ToArray();
 
+        var msbuildArgs = MSBuildArgs.AnalyzeMSBuildArguments([..forwardedOptions, ..binLogArgs], CommonOptions.PropertiesOption, CommonOptions.RestorePropertiesOption, BuildCommandParser.TargetOption);
```

Comment:
Should binlog arg parsing be also part of the new `MSBuildArgs` utility rather than `LoggerUtility.SeparateBinLogArguments`?

Reply:
it definitely could! this PR was already pretty big though so I figured this would be an OK stopping point.